### ISSUE TITLE
Hexadecimal number > 0xffffffff non-portable at Image-ExifTool-11.26/lib/Image/ExifTool/Writer.pl line 3223.

### ISSUE THREAD

**Issue by RealDolos** [exiftool/exiftool#6]
When running exiftool-11.26, fixing/redoing some jpegs, exitool (or rather perl on it's behalf) sometimes spews this warning:
```
Hexadecimal number > 0xffffffff non-portable at Image-ExifTool-11.26/lib/Image/ExifTool/Writer.pl line 3223.
```
In my case, it seems to be a minor cosmetic annoyance (only a warning in 64-bit perl), however I suspect on 32-bit systems it might be an actual problem.

I'm using perl on Ubuntu 18.04 with the official version retrieved from the exiftool homepage.
```txt
This is perl 5, version 26, subversion 1 (v5.26.1) built for x86_64-linux-gnu-thread-multi
(with 67 registered patches, see perl -V for more detail)
```

The command line I was running (fix, remove thumbnail(s), properly set the exif serial numbers, if any, rewrite all + icc; based on the faqs on exiftool):
```sh
./exiftool -F -ifd1:all= -PreviewImage= '-exif:SerialNumber<SerialNumber' -overwrite_original -all= -tagsfromfile @ -all:all -unsafe -icc_profile  -ext jpg -r -q -q $HOME/cam-repair
```

Everything, besides those warning, seems to be fine.

**Comment by boardhead** [comment]
Odd.  I get this:

`% perl -we 'print hex 0x5f5e0ff'
2576980377
% perl -we 'print hex 0x5f5e100'
Hexadecimal number > 0xffffffff non-portable at -e line 1.`

So Perl warns for numbers greater than 0x5f5e0ff.  This would seem to be a bug in Perl.  Even 0xffffffff should be portable because this is 64-bit.  The warning itself indicates that only > 0xffffffff is a problem, yet Perl warns for numbers as small as 0x5f5e100.  I'll add a patch to suppress this warning.

**Comment by RealDolos** [comment]
```sh
$ perl -we 'print hex 0x5f5e100'
Hexadecimal number > 0xffffffff non-portable at -e line 1.
4294967296
$ perl -we 'print hex "0x5f5e100"'
100000000
$ perl -we 'print hex 100000000'
Hexadecimal number > 0xffffffff non-portable at -e line 1.
4294967296
$ perl -we 'print hex "100000000"'
Hexadecimal number > 0xffffffff non-portable at -e line 1.
4294967296
$ perl -we 'print hex "0x100000000"'
Hexadecimal number > 0xffffffff non-portable at -e line 1.
4294967296
```

You did not actually test a string literal, but a number, which perl will then coerce back into a string, but in decimal representation, of course ;)

**Comment by boardhead** [comment]
Right.  That makes sense now.  Thanks.

**Comment by boardhead** [comment]
Resolved by ExifTool 11.29
